tables_io/types.py

```diff
@@ -92,14 +102,19 @@ def tableType(obj):
         return AP_TABLE
     if isinstance(obj, pd.DataFrame):
         return PD_DATAFRAME
+    if isinstance(obj, np.recarray):
+        return NUMPY_RECARRAY
     if not isinstance(obj, Mapping):
         raise TypeError("Object of type %s is not one of the supported types %s" %
                         (type(obj), list(TABULAR_FORMAT_NAMES.keys())))
 
     nRow = None
     for key, val in obj.items():
         if isinstance(val, (Mapping, apTable.Table, pd.DataFrame)):
-            raise TypeError("Column %s of type a Mapping %s" %
+            raise TypeError("Column %s of type Mapping %s" %
+                            (key, type(val)))
+        if isinstance(val, (np.recarray)):
```

Comment:
Yes.   I had thought about this a bit.   What I'm actually doing here is catching cases where someone passed a Mapping of tables to this function, and saying that those aren't tables. 
The issues with which types are acceptable is that in principle I think that any Iterable is ok, as numpy will just iterate over it and turn it into an array when it needs to.   Maybe I'm wrong about that.  Thoughts?

Reply:
If any iterable is ok with numpy (sounds right to me but I haven't looked into it), why specifically exclude a couple things?  For a trivial example - what if I define a class which behaves just like a recarray but isn't a recarray (e.g. method names are different).  Should a mapping of those be excluded?   Once a routine is public, there is no telling what arguments will be passed in.
Maybe what I'm asking is - why exclude these cases?  If it's because it's *probably* a mistake on the caller's part, maybe replace the error with a warning. Or is something else going on that I'm missing?

Reply:
So, this function is doing double duty,  one way in which it is called is to determine if something is a "TableDictLike" object, i.e.,a Mapping of str -> Tablelike.   But, the numpy dict "TableLike" representation is also a Mapping.   So if someone passes in a TableDictLike object, we want this to fail.    Thus we have to watch out for people passing in a Mapping -> str , (stuff that looks like a table).   Maybe it would be clearer if the line has   if isTablelike(val): instead of the current version.  Thoughts?

Reply:
Ok, I switched it to check isTablelike() as that seems like a clearer & more specific check.